javav2/example_code/pinpoint/src/main/java/com/example/pinpoint/CreateEndpoint.java

```diff
@@ -0,0 +1,154 @@
+
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.pinpoint.PinpointClient;
+import software.amazon.awssdk.services.pinpoint.model.*;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+//snippet-end:[pinpoint.java2.CreateEndpoint.import]
+
+public class CreateEndpoint {
+    public static void main(String[] args) {
+
+        final String USAGE = "\n" +
+                "CreateEndpoint - create an endpoint for an application in pinpoint\n\n" +
+                "Usage: CreateEndpoint <appId>\n\n" +
+                "Where:\n" +
+                "  appId - the ID of the application to create an endpoint for.\n\n";
+
+        if (args.length < 1) {
+            System.out.println(USAGE);
+            System.exit(1);
+        }
+        //snippet-start:[pinpoint.java2.CreateEndpoint.main]
+        String appId = args[0];
+
+        String endpointId = UUID.randomUUID().toString();
+        System.out.println("Endpoint ID: " + endpointId);
+
+        PinpointClient pinpoint = PinpointClient.builder().region(Region.US_EAST_1).build();
+
+        EndpointResponse response = createEndpoint(pinpoint, appId);
+
+        System.out.println(response.address());
+        System.out.println(response.channelType());
+        System.out.println(response.applicationId());
+        System.out.println(response.endpointStatus());
+        System.out.println(response.requestId());
+        System.out.println(response.user());
+        //snippet-end:[pinpoint.java2.CreateEndpoint.main]
+    }
+
+    //snippet-start:[pinpoint.java2.CreateEndpoint.helper]
+    private static EndpointResponse createEndpoint(PinpointClient client, String appId) {
+        String endpointId = UUID.randomUUID().toString();
+        System.out.println("Endpoint ID: " + endpointId);
+
+        EndpointRequest endpointRequest = createEndpointRequestData();
+
+
+        UpdateEndpointRequest updateEndpointRequest = UpdateEndpointRequest.builder()
+                .applicationId(appId)
+                .endpointId(endpointId)
+                .endpointRequest(endpointRequest)
+                .build();
+
+        UpdateEndpointResponse updateEndpointResponse = client.updateEndpoint(updateEndpointRequest);
+        System.out.println("Update Endpoint Response: " + updateEndpointResponse.messageBody());
+
+        GetEndpointRequest getEndpointRequest = GetEndpointRequest.builder()
+                .applicationId(appId)
+                .endpointId(endpointId)
+                .build();
+        GetEndpointResponse getEndpointResponse = client.getEndpoint(getEndpointRequest);
+
+        System.out.println("Got Endpoint: " + getEndpointResponse.endpointResponse().id());
+        return getEndpointResponse.endpointResponse();
+
+    }
+
+    private static EndpointRequest createEndpointRequestData() {
+        HashMap<String, List<String>> customAttributes = new HashMap<>();
```

Comment:
Move this line before line 104, its more clear to create and update variables together